src/components/join-flow/join-flow.jsx

```diff
@@ -121,21 +121,27 @@ class JoinFlow extends React.Component {
     //   * "birth_month": ["Ensure this value is greater than or equal to 1."]
     handleRegistrationResponse (err, body, res) {
         this.setState({
-            numAttempts: this.state.numAttempts + 1,
-            waiting: false
+            numAttempts: this.state.numAttempts + 1
```

Comment:
we must not set waiting to false yet -- not until we know the session retries are over.